scripts/cmake/vcpkg_configure_make.cmake

```
@@ -73,300 +53,453 @@
-            string(REPLACE "/" "-" TMP_CFLAGS "${TMP_CFLAGS}")
+        set(TAR_DIR "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-dbg")
+        file(MAKE_DIRECTORY "${TAR_DIR}")
+        file(RELATIVE_PATH RELATIVE_BUILD_PATH "${TAR_DIR}" "${SRC_DIR}")
+
+        if(_csc_COPY_SOURCE)
+            file(COPY "${SRC_DIR}/" DESTINATION "${TAR_DIR}")
+            set(RELATIVE_BUILD_PATH .)
+        endif()
+    
+        if(ENV{PKG_CONFIG_PATH})
+            set(BACKUP_ENV_PKG_CONFIG_PATH_DEBUG $ENV{PKG_CONFIG_PATH})
+            set(ENV{PKG_CONFIG_PATH} "${CURRENT_INSTALLED_DIR}/debug/lib/pkgconfig${VCPKG_HOST_PATH_SEPARATOR}${CURRENT_INSTALLED_DIR}/share/pkgconfig${VCPKG_HOST_PATH_SEPARATOR}$ENV{PKG_CONFIG_PATH}")
+        else()
+            set(ENV{PKG_CONFIG_PATH} "${CURRENT_INSTALLED_DIR}/debug/lib/pkgconfig${VCPKG_HOST_PATH_SEPARATOR}${CURRENT_INSTALLED_DIR}/share/pkgconfig")
+        endif()
+        # Setup debug enviromnent
+        if (CMAKE_HOST_WIN32) # Flags should be set in the toolchain instead
+            set(TMP_CFLAGS "${C_FLAGS_GLOBAL} ${VCPKG_CRT_LINK_FLAG_PREFIX}d /D_DEBUG /Ob0 /Od ${VCPKG_C_FLAGS_DEBUG}")
+            string(REGEX REPLACE "[ \t]+/" " -" TMP_CFLAGS "${TMP_CFLAGS}")
             set(ENV{CFLAGS} ${TMP_CFLAGS})
-            set(TMP_CXXFLAGS "${CXX_FLAGS_GLOBAL} ${CMAKE_CXX_FLAGS} ${CMAKE_CXX_FLAGS_DEBUG}")
-            string(REPLACE "/" "-" TMP_CXXFLAGS "${TMP_CXXFLAGS}")
+            
+            set(TMP_CXXFLAGS "${CXX_FLAGS_GLOBAL} ${VCPKG_CRT_LINK_FLAG_PREFIX}d /D_DEBUG /Ob0 /Od ${VCPKG_CXX_FLAGS_DEBUG}")
+            string(REGEX REPLACE "[ \t]+/" " -" TMP_CXXFLAGS "${TMP_CXXFLAGS}")
             set(ENV{CXXFLAGS} ${TMP_CXXFLAGS})
-            set(TMP_LDFLAGS "${LD_FLAGS_GLOBAL}")
-            string(REPLACE "/" "-" TMP_LDFLAGS "${TMP_LDFLAGS}")
+            
+            set(TMP_LDFLAGS "${LD_FLAGS_GLOBAL} -L${_VCPKG_INSTALLED}/debug/lib ${VCPKG_LINKER_FLAGS_DEBUG}")
+            string(REGEX REPLACE "[ \t]+/" " -" TMP_LDFLAGS "${TMP_LDFLAGS}")
             set(ENV{LDFLAGS} ${TMP_LDFLAGS})
-        endif()
-        
-        set(OBJ_DIR ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-dbg)
-        set(PRJ_DIR ${OBJ_DIR}/${_csc_PROJECT_SUBPATH})
-        
-        file(MAKE_DIRECTORY ${OBJ_DIR})
-        
-        if (NOT CMAKE_HOST_WIN32)
-            file(GLOB_RECURSE SOURCE_FILES ${_csc_SOURCE_PATH}/*)
-            foreach(ONE_SOUCRCE_FILE ${SOURCE_FILES})
-                get_filename_component(DST_DIR ${ONE_SOUCRCE_FILE} PATH)
-                string(REPLACE "${_csc_SOURCE_PATH}" "${OBJ_DIR}" DST_DIR "${DST_DIR}")
-                file(COPY ${ONE_SOUCRCE_FILE} DESTINATION ${DST_DIR})
-            endforeach()
+            set(ENV{PKG_CONFIG} "${PKGCONFIG} --define-variable=prefix=${_VCPKG_INSTALLED}/debug")
+            
+            set(dbg_command
+                ${base_cmd} -c "${CONFIGURE_ENV} ./${RELATIVE_BUILD_PATH}/configure ${BUILD_TARGET} ${HOST_TYPE}${_csc_OPTIONS} ${_csc_OPTIONS_DEBUG}")

```

```suggestion
                ${base_cmd} -c "${CONFIGURE_ENV} ./${SRC_DIR}/configure ${BUILD_TARGET} ${HOST_TYPE}${_csc_OPTIONS} ${_csc_OPTIONS_DEBUG}")

```
/.. **/./** ../ - error **perl** search path \\ and in other similar cases
SRC_DIR work path

---

Whats exactly the problem here or when exactly does the problem occur (so that i can check it)? I use `RELATIVE_BUILD_PATH` because it is certain to not have any spaces in it while `SRC_DIR` could possible have spaces screwing up the call.

---

https://github.com/microsoft/vcpkg/pull/10402#issuecomment-599155720

see error in `config.log`

---

this can be seen during any compilation, but the error will only be returned if additional utilities such as PERL are used

> это можно увидеть при любой компиляции, но ошибку выдаст только если используется дополнительные утилиты, типа PERL

---

@Voskrese: I don't know how to reproduce your error. And I tried with #10402 (comment) but ran into other issues with the mentioned portfiles. 
Please add the `<port>/CONTROL` and `<port>/portfile.cmake` you used so that I can have a look on the errors